dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/export/relationship/RelationshipOperationParamsMapper.java

```diff
@@ -73,30 +85,47 @@ public RelationshipQueryParams map(RelationshipOperationParams params) throws No
         .build();
   }
 
-  private TrackedEntity validateTrackedEntity(String uid) throws NotFoundException {
+  private TrackedEntity validateTrackedEntity(User user, String uid)
+      throws NotFoundException, ForbiddenException {
     TrackedEntity trackedEntity = trackedEntityService.getTrackedEntity(uid);
     if (trackedEntity == null) {
       throw new NotFoundException(TrackedEntity.class, uid);
     }
 
+    List<String> errors = accessManager.canRead(user, trackedEntity);
+    if (!errors.isEmpty()) {
+      throw new ForbiddenException("User has no access to TrackedEntity: " + uid);
```

Comment:
I would prefer if we would write sentences like

```suggestion
      throw new ForbiddenException("User has no access to tracked entity: " + uid);
```

and `throw new ForbiddenException("User has no access to enrollment: " + uid);`. We sometimes [write this way](https://github.com/dhis2/dhis2-core/blob/d848f6875c5d49f5697010e2ff0ba7dba07d2bae/dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/export/event/EventOperationParamsMapper.java#L279).

We not consistent yet. First, we should settle on one of these ways of writing 😅

Reply:
Yeah, I do not like to write errors. Check if you like the fix, following the same pattern as `NotFoundException`